ingest/usecase/process_alloyed_pool.go

```diff
@@ -0,0 +1,89 @@
+// CONTRACT: the caller checked that this is an alloyed pool
+func processAlloyedPool(sqsModel *sqsdomain.SQSPool) error {
+	if len(sqsModel.CosmWasmPoolModel.Data.AlloyTransmuter.AssetConfigs) == 0 {
+		return fmt.Errorf("no asset configs found for alloyed pool")
+	}
+
+	cosmWasmModel := sqsModel.CosmWasmPoolModel
+
+	// Compute the standard normalization factor
+	standardNormalizationFactor, err := computeStandardNormalizationFactor(cosmWasmModel.Data.AlloyTransmuter.AssetConfigs)
+	if err != nil {
+		return err
+	}
+
+	// Compute the scaling factor normalization factors from the standard normalization factor for each asset
+	normalizationScalingFactors, err := computeNormalizationScalingFactors(standardNormalizationFactor, cosmWasmModel.Data.AlloyTransmuter.AssetConfigs)
+	if err != nil {
+		return err
+	}
+
+	// Update the precomputed data in the model
+	cosmWasmModel.Data.AlloyTransmuter.PreComputedData.StdNormFactor = standardNormalizationFactor
+	cosmWasmModel.Data.AlloyTransmuter.PreComputedData.NormalizationScalingFactors = normalizationScalingFactors
+
+	return nil
+}
+
+// computeStandardNormalizationFactor computes the standard normalization factor for the pool.
+// Returns error if one of the asset normalization factors is nil or zero.
+func computeStandardNormalizationFactor(assetConfigs []cosmwasmpool.TransmuterAssetConfig) (osmomath.Int, error) {
+	result := osmomath.OneInt().BigIntMut()
+	for i := 0; i < len(assetConfigs); i++ {
+		normFactor := assetConfigs[i].NormalizationFactor
+		if normFactor.IsNil() || normFactor.IsZero() {
+			return osmomath.Int{}, fmt.Errorf("normalization factor is nil or zero for asset %s", assetConfigs[i])
+		}
+
+		currentNormFactor := assetConfigs[i].NormalizationFactor.BigInt()
+		result = Lcm(result, currentNormFactor)
+	}
+	return osmomath.NewIntFromBigInt(result), nil
+}
+
+// computeNormalizationScalingFactors computes the normalization scaling factors for each denom in the asset config
+// using the standard normalization factor.
+// Returns error if one of the asset normalization factors is nil or zero.
+// Returns error if the standard normalization factor is nil or zero.
+// Returns error if asset scaling factor truncates to zero.
+func computeNormalizationScalingFactors(standardNormalizationFactor osmomath.Int, assetConfigs []cosmwasmpool.TransmuterAssetConfig) ([]osmomath.Int, error) {
+	if standardNormalizationFactor.IsNil() || standardNormalizationFactor.IsZero() {
+		return nil, fmt.Errorf("standard normalization factor is nil or zero")
+	}
+
+	scalingFactors := make([]osmomath.Int, len(assetConfigs))
+	for i := 0; i < len(assetConfigs); i++ {
+		assetNormalizationFactor := assetConfigs[i].NormalizationFactor
+		if assetNormalizationFactor.IsNil() || assetNormalizationFactor.IsZero() {
+			return nil, fmt.Errorf("normalization factor is nil or zero for asset %s", assetConfigs[i].Denom)
+		}
+
+		assetScalingFactor := standardNormalizationFactor.Quo(assetNormalizationFactor)
```

Comment:
@iboss-ptk keep in mind that I pre-compute this value at the time of ingest

Reply:
I see, it's precomputed on the ingest time on sqs side, not from node.

Reply:
Oh get it now